lib/java/com/google/android/material/internal/CollapsingTextHelper.java

```diff
@@ -447,6 +449,27 @@ public void apply(Typeface font) {
     recalculate();
   }
 
+  public void setTitleTextEllipsize(int ellipsize) {
+    // Convert to supported TextUtils.TruncateAt values
+    switch (ellipsize) {
+      case 0:
+        titleTextEllipsize = TextUtils.TruncateAt.START;
+        break;
+      case 1:
+        titleTextEllipsize = TextUtils.TruncateAt.MIDDLE;
+        break;
+      case 2:
+        titleTextEllipsize = TextUtils.TruncateAt.END;
+        break;
+      case 3:
+        titleTextEllipsize = TruncateAt.MARQUEE;
+        break;
+      default:
+        titleTextEllipsize = TextUtils.TruncateAt.END;
+        break;
+    }
```

Comment:
I think we need to call recalculate() here. And in calculateUsingTextSize we need to make sure textLayout will be recreated if titleTextEllipsize changes.

Reply:
@drchen  updated, Please check the latest changes